src/lib/mina_graphql/mina_graphql.ml

```diff
@@ -4619,8 +4626,27 @@ module Mutations = struct
                                 (Public_key.compress public_key, private_key) )
                             |> Public_key.Compressed.Map.of_alist_exn
                           in
-                          let rec go ~vk ~prover account_state_tbl ndx tm_next
-                              counter =
+                          let insert_recently_used_accounts
+                              ~recently_used_accounts ~account_state_tbl id =
+                            let a =
+                              Account_id.Table.find_and_remove account_state_tbl
+                                id
+                            in
+                            Queue.enqueue recently_used_accounts
+                              (Option.value_exn a) ;
+                            if
+                              Queue.length recently_used_accounts
+                              > zkapp_command_details.recently_used_accounts
+                            then
+                              let a, role =
+                                Queue.dequeue_exn recently_used_accounts
+                              in
+                              Account_id.Table.add_exn account_state_tbl
+                                ~key:(Account.identifier a) ~data:(a, role)
+                            else ()
+                          in
+                          let rec go ~vk ~prover ~account_state_tbl ~ndx
```

Comment:
`recently_used_accounts` changes by side-effect, I believe, so doesn't need to be passed around the `go` loop

Reply:
you could define it somewhere above `go`

Reply:
the same is true for `account_state_tbl`